x-pack/plugins/maps/public/classes/sources/es_geo_grid_source/es_geo_grid_source.js

```diff
@@ -282,6 +289,7 @@ export class ESGeoGridSource extends AbstractESAggSource {
             precision: searchFilters.geogridPrecision,
             layerName,
             registerCancelCallback,
+            bufferedExtent: searchFilters.buffer,
```

Comment:
Can not use searchFilters.buffer. Need to expand searchFilters.buffer to tile boundaries so it sits on tile evenly.

Here is an example where the buffer is expanded for the filter portion of request DSL
https://github.com/elastic/kibana/blob/master/x-pack/plugins/maps/public/classes/sources/es_source/es_source.js#L120
```
const buffer = this.isGeoGridPrecisionAware()
        ? expandToTileBoundaries(searchFilters.buffer, searchFilters.geogridPrecision)
        : searchFilters.buffer;
```

This was done to fix https://github.com/elastic/kibana/issues/54038

Reply:
discussed in person. Will leave as-is. The nested-bounds makes a selection on the bucket-level while the snapped-bounds on the filter ensures all documents in a given grid-cell are included for the aggregation in that cell. The nested-bounds would always capture the required buckets.